src/Development/IDE/Types/Location.hs

```diff
@@ -10,151 +10,64 @@ module Development.IDE.Types.Location
     , Position(..)
     , showPosition
     , Range(..)
-    , Uri(..)
-    , NormalizedUri
+    , LSP.Uri(..)
+    , LSP.NormalizedUri
     , LSP.toNormalizedUri
     , LSP.fromNormalizedUri
-    , NormalizedFilePath
+    , NormalizedFilePath'
     , fromUri
-    , toNormalizedFilePath
-    , fromNormalizedFilePath
-    , filePathToUri
+    , toNormalizedFilePath'
+    , fromNormalizedFilePath'
     , filePathToUri'
     , uriToFilePath'
     , readSrcSpan
     ) where
 
 import Control.Applicative
 import Language.Haskell.LSP.Types (Location(..), Range(..), Position(..))
-import Control.DeepSeq
 import Control.Monad
-import Data.Binary
-import Data.Maybe as Maybe
-import Data.Hashable
+import Data.Binary (Binary)
+import Control.DeepSeq (NFData)
+import Data.Hashable (Hashable, hash)
 import Data.String
-import qualified Data.Text as T
 import FastString
-import Network.URI
-import System.FilePath
-import qualified System.FilePath.Posix as FPP
-import qualified System.FilePath.Windows as FPW
-import System.Info.Extra
 import qualified Language.Haskell.LSP.Types as LSP
-import Language.Haskell.LSP.Types as LSP (
-    filePathToUri
-  , NormalizedUri(..)
-  , Uri(..)
-  , toNormalizedUri
-  , fromNormalizedUri
-  )
 import SrcLoc as GHC
 import Text.ParserCombinators.ReadP as ReadP
-import GHC.Generics
 
+-- We need a newtype to override the IsString instance
+newtype NormalizedFilePath' = NormalizedFilePath' LSP.NormalizedFilePath
```

Comment:
I think it’s better to just kill the `IsString` instance. The few cases where we actually have `""` literals should be easy to detect and having a separate type all over the place seems confusing.

Reply:
Mmm ok, but are you ok with keeping `toNormalizedFilePath'`/`fromNormalizedFilePath'` (with `'`) to take in account the empty path case?

Reply:
Yes, those are harder to get rid off since the input is a variable rather than a literal so I think it makes sense to keep them.